server/data/prisonApiClient.ts

```diff
@@ -1,46 +1,42 @@
-import RestClient from './restClient'
-import { PrisonApiClient } from './interfaces/prisonApiClient'
+import { AuthenticationClient } from '@ministryofjustice/hmpps-auth-clients'
+import { asUser, RestClient } from '@ministryofjustice/hmpps-rest-client'
 import { CaseLoad } from './interfaces/caseLoad'
 import { Location } from './interfaces/location'
+import { LatestArrivalDateInfo } from './interfaces/latestArrivalDateInfo'
+import logger from '../../logger'
+import config from '../config'
 import PrisonRollCount from './interfaces/prisonRollCount'
 import EstablishmentRollSummary from '../services/interfaces/establishmentRollService/EstablishmentRollSummary'
-import { LatestArrivalDateInfo } from './interfaces/latestArrivalDateInfo'
-
-export default class PrisonApiRestClient implements PrisonApiClient {
-  constructor(private restClient: RestClient) {}
-
-  private get<T>(args: object): Promise<T> {
-    return this.restClient.get<T>(args)
-  }
-
-  private put<T>(args: object): Promise<T> {
-    return this.restClient.put<T>(args)
-  }
+import { PrisonApiClient } from './interfaces/prisonApiClient'
 
-  private async post<T>(args: object): Promise<T> {
-    return this.restClient.post<T>(args)
+export default class PrisonApiRestClient extends RestClient implements PrisonApiClient {
+  constructor(authenticationClient: AuthenticationClient) {
+    super('Prison API', config.apis.prisonApi, logger, authenticationClient)
   }
 
-  getUserCaseLoads(): Promise<CaseLoad[]> {
-    return this.get<CaseLoad[]>({ path: '/api/users/me/caseLoads', query: 'allCaseloads=true' })
+  getUserCaseLoads(token: string): Promise<CaseLoad[]> {
+    return this.get<CaseLoad[]>({ path: '/api/users/me/caseLoads', query: 'allCaseloads=true' }, asUser(token))
   }
 
-  getUserLocations(): Promise<Location[]> {
-    return this.get<Location[]>({ path: '/api/users/me/locations' })
+  getUserLocations(token: string): Promise<Location[]> {
+    return this.get<Location[]>({ path: '/api/users/me/locations' }, asUser(token))
   }
 
-  setActiveCaseload(caseLoad: CaseLoad): Promise<Record<string, string>> {
-    return this.put<Record<string, string>>({ path: '/api/users/me/activeCaseLoad', data: caseLoad })
+  setActiveCaseload(token: string, caseLoad: CaseLoad): Promise<Record<string, string>> {
+    return this.put<Record<string, string>>({ path: '/api/users/me/activeCaseLoad', data: caseLoad }, asUser(token))
   }
 
-  getPrisonRollCountSummary(prisonId: string): Promise<EstablishmentRollSummary> {
-    return this.get<PrisonRollCount>({ path: `/api/prison/roll-count/${prisonId}/summary` })
+  getPrisonRollCountSummary(token: string, prisonId: string): Promise<EstablishmentRollSummary> {
+    return this.get<PrisonRollCount>({ path: `/api/prison/roll-count/${prisonId}/summary` }, asUser(token))
```

Comment:
Most of these we don't want to be user tokens - better to keep these as raw tokens